src/clj_sqs_extended/core.clj

```diff
@@ -0,0 +1,110 @@
+(ns clj-sqs-extended.core
+  "Provides the core functionalities of the wrapped library."
+  (:require [clj-sqs-extended.tools :as tools])
+  (:import [com.amazonaws.services.s3 AmazonS3ClientBuilder]
+           [com.amazonaws.services.s3.model ListVersionsRequest]
+           [com.amazonaws.services.sqs AmazonSQSClientBuilder]
+           [com.amazon.sqs.javamessaging
+              AmazonSQSExtendedClient
+              ExtendedClientConfiguration]
+           [com.amazonaws.services.sqs.model
+              ReceiveMessageRequest
+              DeleteMessageRequest]
+           [com.amazonaws.client.builder AwsClientBuilder$EndpointConfiguration]
+           [com.amazonaws.services.s3.model
+              BucketLifecycleConfiguration
+              BucketLifecycleConfiguration$Rule]))
+
+
+(defn configure-endpoint
+  "Creates an endpoint configuration for the passed url and region."
+  [url region]
+  (AwsClientBuilder$EndpointConfiguration. url region))
+
+(defn configure-bucket-lifecycle
```

Comment:
some of these fns in this ns are not expected to be part of functionality of this library, move them into `test/`. limit code in `src/` down to the esssential.